integration/disk_cleanup_deletes_inactive_directories_test.go

```diff
@@ -75,7 +76,29 @@ func TestDiskCleansupInactiveFilesets(t *testing.T) {
 	require.NoError(t, err)
 	testSetup.db.AssignShardSet(shardSet)
 
+	// So some gnarly stuff here to make the testing package a bit more synchronous
+	var fsCleanupErr chan error
+	var nsCleanupErr chan error
 	// Check if files have been deleted
 	waitTimeout := 30 * time.Second
-	require.NoError(t, waitUntilFilesetsCleanedUp(filePathPrefix, md.ID(), extraShard.ID(), waitTimeout))
+
+	go func() {
+		fsCleanupErr <- waitUntilFilesetsCleanedUp(filePathPrefix,
+			testSetup.db.Namespaces(), extraShard.ID(), waitTimeout)
+		require.NoError(t, <-fsCleanupErr)
+	}()
+
+	// Assigning a shardset of length one will mean there are extra namespaces
+	// that will need to be removed via cleanup
+	shardSet, err = sharding.NewShardSet(shards[:1], shardSet.HashFn())
+	require.NoError(t, err)
+	testSetup.db.AssignShardSet(shardSet)
+
+	// Needs an extra 30 seconds due to race conditions
+	waitTimeout = 160 * time.Second
```

Comment:
Better to use two different variables to pass to the different goroutines rather than modifying this at the top level, it's not guarenteed they won't read them in a non-racy way.